javabridge/locate.py

```diff
@@ -15,14 +15,39 @@
 import logging
 import subprocess
 
+
+# Note: if a matching gcc is available from the shell on Windows, its
+#       probably safe to assume the user is in an MINGW or MSYS or Cygwin
+#       environment, in which case he/she wants to compile with gcc for
+#       Windows, in which case the correct compiler flags will be triggered
+#       by is_mingw. This method is not great, improve it if you know a
+#       better way to discriminate between compilers on Windows.
+def is_mingw():
+    if (os.name=="nt"):
+        mingw32 = ""
```

Comment:
I would put a check in here to see if the environment variables, MSSdk and DISTUTILS_USE_SDK are set to values other than zero (see https://github.com/cython/cython/wiki/CythonExtensionsOnWindows). That's an indication of the  user's intent - to compile using the Microsoft toolchain.